contracts/Content/UniqueContent.sol

```diff
@@ -1,40 +1,52 @@
 import "@openzeppelin/contracts-upgradeable/interfaces/IERC2981Upgradeable.sol";
-import "../libraries/LibAsset.sol";
-import "./MultipleRoyalties.sol";
 import "./interfaces/IMultipleRoyalties.sol";
 import "./interfaces/IUniqueContent.sol";
+import "./interfaces/IUniqueContentStorage.sol";
 import "./interfaces/IContent.sol";
+import "../libraries/LibRoyalty.sol";
 
-contract UniqueContent is IUniqueContent, IMultipleRoyalties, MultipleRoyalties, ERC721Upgradeable, ERC1155HolderUpgradeable, IERC2981Upgradeable {
+contract UniqueContent is IUniqueContent, IMultipleRoyalties, ERC721Upgradeable, ERC1155HolderUpgradeable, IERC2981Upgradeable, ERC165StorageUpgradeable {
 
     /***************** Stored Variables *****************/
-    mapping(uint256 => LibAsset.UniqueAsset) uniqueAssetInfo;
+    IUniqueContentStorage uniqueContentStorage;
     uint256 private uniqueIdsCounter;
+
+    function initialize(
+        address _uniqueContentStorage)
+        public initializer
+    {
+        _registerInterface(type(IMultipleRoyalties).interfaceId);
+        _registerInterface(type(IERC2981Upgradeable).interfaceId);
+
+        uniqueContentStorage = IUniqueContentStorage(_uniqueContentStorage);
+    }
     
     /** Asset Minting
     * @dev If the royalties are valid and if the caller has the original item in their wallet, it takes the original asset, mints the unique asset and updates mappings
     * @param _data LibAsset.UniqueAssetCreateData structure object
     */
     function mint(LibAsset.UniqueAssetCreateData memory _data) external override {
-        require((IERC1155Upgradeable(_data.contentAddress).balanceOf(_msgSender(), _data.tokenId)) >= 1, "You must have the original item in your wallet");
+        require(
+            (ERC165CheckerUpgradeable.supportsInterface(_data.contentAddress, type(IERC2981Upgradeable).interfaceId)) &&
+            (ERC165CheckerUpgradeable.supportsInterface(_data.contentAddress, type(IERC1155Upgradeable).interfaceId)),
+            "Error: content contract not supported"
+        );
+        require((IERC1155Upgradeable(_data.contentAddress).balanceOf(_msgSender(), _data.tokenId)) >= 1, "Error: must have original item");
         (, uint256 _originalRoyaltyRate) = IERC2981Upgradeable(_data.contentAddress).royaltyInfo(_data.tokenId, 1e6);
-        require(_verifyRoyalties(_data.royaltyReceivers, _data.royaltyRates, _originalRoyaltyRate), "The royalties entered are invalid");
+        require(LibRoyalty.verifyRoyalties(_data.royaltyReceivers, _data.royaltyRates, _originalRoyaltyRate), "Error: royalties are invalid");
         // transfers the original asset to be locked in the unique content contract
         IERC1155Upgradeable(_data.contentAddress).safeTransferFrom(_msgSender(), address(this), _data.tokenId, 1, "");
         // mint() is a mint and transfer function, if _data.to != msgSender, the caller would be sending the token to someone else
         _mint(_data.to, uniqueIdsCounter);
```

Comment:
Also, I don't see uniqueIdsCounter ever incremented. Where is this incremented?

Reply:
it increments after the emit event

Reply:
oh, oops, it was minimized in the code review. 

With that said, it doesn't need to be in it's own line. You can just do

emit Mint(uniqueIdsCounter++, _msgSender(), _data);

This also increments after uniqueIdsCounter is sent along with the Mint event